src/runtime/virtcontainers/kata_agent.go

```diff
@@ -2181,40 +2182,57 @@ func (k *kataAgent) setGuestDateTime(ctx context.Context, tv time.Time) error {
 func (k *kataAgent) copyFile(ctx context.Context, src, dst string) error {
 	var st unix.Stat_t
 
-	err := unix.Stat(src, &st)
+	err := unix.Lstat(src, &st)
```

Comment:
This is a change in behaviour: we used to just copy the contents of the pointee, now we're send the link in (potentially to a non-existing file).

Are we sure this isn't break anything?

Reply:
Yes, it changes the original behavior, but I am not sure it break any existing use cases.

One idea to keep the compatibility is that we de-reference the root of the source path.  The original implementation does not support recursive copying, so the source path is a regular file or a symlink to a regular file.

We use `unix.Stat` for the root of the source path, and determine whether it is a file or directory. If the source path is a directory, then its children are examined using `unix.Lstat`.